golem-worker-executor-base/src/durable_host/golem/v1x.rs

```diff
@@ -255,6 +257,112 @@ impl<Ctx: WorkerCtx> Host for DurableWorkerCtx<Ctx> {
 
         Ok(())
     }
+
+    async fn resolve_component_id(
+        &mut self,
+        component_slug: String,
+    ) -> anyhow::Result<Option<ComponentId>> {
+        let durability =
+            Durability::<Option<golem_common::model::ComponentId>, SerializableError>::new(
+                self,
+                "golem::api",
+                "resolve_component_id",
```

Comment:
There is a (or maybe two) big match on these strings where extra information is constructed for oplog query results (`PublicOplogEntry`). Please add the new case(s) there.

Reply:
(My plan is that once we break the oplog format we will make these serialized values self-describing so that manual mapping won't be necessary)